packages/main/src/StepInput.ts

```diff
@@ -366,8 +366,16 @@ class StepInput extends UI5Element implements IFormElement {
 		return this.focused;
 	}
 
-	get _valuePrecisioned() {
-		return this.value.toFixed(this.valuePrecision);
+	get _displayValue() {
+		if ((this.value === 0) || (Number.isInteger(this.value) && !this.valuePrecision)) {
```

Comment:
It is not correct to return `this.value` at the end as it is of type `number`, while this getter is intended to return a string representation of the numeric value.

As of the check it is intended to check if the number is integer rather than decimal, and the second part of the check is just to secure the check. Because for example the number typed in the input could be `1000.00` while the `valuePrecision` is 0/not provided. In that case the `Number.isInteger(this.value)` will return true and it will round the value with `Number.toFixed()`, which we want to avoid atm.